### ISSUE TITLE
Add support for Discord

### ISSUE THREAD

**Issue by HDoujinDownloader** [HDoujinDownloader/HDoujinDownloader#354]
A user has requested support for downloading media from Discord channels. 

I'd definitely like to add support for this, but it has the same challenges as downloading from booru websites. HDoujin Downloader currently enumerates all page URLs before starting the download, which is infeasible for galleries with a potentially very large number of images requiring very many requests to enumerate.

I'll probably add a means for `BeforeDownloadPage` to modify the page list and insert new image URLs while the download is running. This introduces some challenges, though, such as how to handle resuming a partial download.

**Comment by ale5000-git** [comment]
It would be nice to have the possibility to set an imprecise/guessed page count when it is added to the list of the app, and then correct it to the real value when the download start.
Is it possible?

**Comment by HDoujinDownloader** [comment]
Yes: If you're completely unsure how many pages there are, you can set the page count to `'?'`, which is updated to be actual page count when the pages are fetched. You can also set it to whatever approximation you want and have it updated later if you add `module.Strict = false` in `Register`. What this does is makes it so there isn't an error if the fetched page count doesn't match the specified one.

That doesn't help address this problem though, because HDoujin Downloader still fetches all the pages before starting the download. I need to change how things work internally so new pages can be fetched while the download is actively running.
